### DISCUSSION TITLE
[Help] How to create and use Resource and how to redirct onEventClick ?

### DISCUSSION THREAD

**Discussion by abdo17rabah** [GuavaCZ/calendar#56] [Q&A] [answered]
I tried to follow the documentation and the demo but couldn't get the result i'm looking for which is showing the appointments of a doctor. I succeeded to show the events without using `Resource`, but in order to use `timeGridWeek ` i need to use it. 

I have a Model `Doctor` which can have many `Appointments`. 

I've implemented `Resourceable`  in `Doctor` as the following : 

```
public function toResource(): array|Resource
    {
        return Resource::make($this->id)
            ->title(Auth()->user()->full_name);
    }
```

In Model `Appointment` i've implemented `Eventable` as the following :

```
public function toEvent(): array|Event
    {
        return Event::make(Appointment::find($this->id))
            ->title($this->reference . ' - ' . $this->patient->full_name)
            ->start($this->start_at)
            ->end($this->end_at)
            ->resourceId(auth()->user()->doctor->id)
            ->action('view');
    }
```

Then in the widget i've defined this : 

```
protected string $calendarView = 'timeGridWeek';
    protected static bool $isLazy = true;

    public function getEvents(array $fetchInfo = []): Collection|array
    {
        return auth()->user()->doctor->appointments()
            ->with(['patient'])
            ->get()
            ->map(fn ($appointment) => $appointment->toEvent())->all();
    }
    public function getResources(): Collection|array
    {
        return Doctor::query()
            ->where('user_id', auth()->id())
            ->get()
            ->map(fn ($doctor) => [
                'id' => $doctor->id,
                'title' => $doctor->name,
            ]);
    }
```

I don't know why the events are appearing in the calendar. 

Also how can i redirect onEventClick on an Event to the `route('filament.app.resources.appointments.create')` for example, what's the correct way? I tried to use this solution but i noticed that it takes time to redirect to the url like there is a latency to get the response:

```
public function onEventClick(array $info = [], ?string $action = null): void
    {
        $this->redirect(route('filament.app.resources.appointments.view', ['record' => $info['event']['extendedProps']['key']]));
    }
```

Thanks for your help.

**Comment by lukas-frey** [comment, marked as answer]
If you want to use resources, you need to use a resource view (their names start with resource)

It takes time because onEventClick is called via an GET request from JavaScript.

Delete the onEventClick method from your class and just add an url extended prop to your event.

The plugin will then use that to redirect immediately without the server roundtrip.

**Reply by abdo17rabah** [reply]
Thanks for your answer.
